tests/integration/main.rs

```diff
@@ -0,0 +1,93 @@
+mod artichoke;
+
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+use serde::ser::{Serialize, Serializer, SerializeStruct};
+use bstr::BString;
+
+#[allow(dead_code)]
+#[derive(Debug)]
+struct CommandOutput {
+    call_args: Vec<String>,
+    status: i32,
+    stdout: BString,
+    stderr: BString,
+}
+
+impl CommandOutput {
+    fn new() -> Self {
+        Self {
+            call_args: vec![],
+            status: -1,
+            stdout: BString::from(""),
+            stderr: BString::from(""),
+        }
+    }
+
+    fn with_args(&mut self, call_args: &Vec<&str>) -> &mut Self {
```

Comment:
Passing `&Vec` and `&String` is not idiomatic. These types coerce to slices which are one word smaller in stack size and have all of the same methods.

```suggestion
    fn with_args(&mut self, call_args: &[&str]) -> &mut Self {
```

Reply:
Magic! I fixed this with the linting errors - since they agree too haha.